windows-active-directory/main.tf

```diff
@@ -0,0 +1,200 @@
+locals {
+  dc1_fqdn = "sampleappdc1.sample.local"
+  
+  dc1_prereq_ad_1 = "Import-Module ServerManager"
+  dc1_prereq_ad_2 = "Install-WindowsFeature AD-Domain-Services -IncludeAllSubFeature -IncludeManagementTools"
+  dc1_prereq_ad_3 = "Install-WindowsFeature DNS -IncludeAllSubFeature -IncludeManagementTools"
+  dc1_prereq_ad_4 = "Import-Module ADDSDeployment"
+  dc1_prereq_ad_5 = "Import-Module DnsServer"
+
+  dc1_install_ad_1 = "Install-ADDSForest -DomainName ad.gpfs.net -DomainNetbiosName dnetname -DomainMode WinThreshold -ForestMode WinThreshold "
+  dc1_install_ad_2 = "-DatabasePath C:/Windows/NTDS -SysvolPath C:/Windows/SYSVOL -LogPath C:/Windows/NTDS -NoRebootOnCompletion:$false -Force:$true "
+  dc1_install_ad_3 = "-SafeModeAdministratorPassword (ConvertTo-SecureString AdminPassword123! -AsPlainText -Force)"
+  
+  dc1_shutdown_command   = "shutdown -r -t 10"
+
+	grafana_prereq_1 =  "Invoke-WebRequest -Uri https://raw.githubusercontent.com/BominRahmani/windows-grafana-template/main/agent-config.yaml -OutFile C:/agent-config.yaml"
```

Comment:
Will this be changed to a file within this repo after the first merge?

Reply:
yea I can do that!

Reply:
Fetching the agent config file from an internet location can be problematic for version control, and in cases where a user chooses to launch their VM in an account with restrictive network access controls.

There are a few different approaches that can be used, which would use a local file, from the host running terraform (where the code is running), in decreasing order of preference.

1. Use the [azurerm_windows_virtual_machine.custom_data](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine#custom_data) argument to create the agent config file.
2. Upload the agent config file to a [blob storage bucket](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_blob), then download it using a web request like you are already doing.
3. Use a VM [file provisioner](https://developer.hashicorp.com/terraform/language/resources/provisioners/file) (not advised)

In each of these cases, you could use a [templatefile](https://developer.hashicorp.com/terraform/language/functions/templatefile) to render the complete agent config, with user parameters, before sending it to the VM or storage location.

Reply:
@rgeyer Sounds good!